pkg/auth/userregistry/identitymapper/provision_test.go

```diff
@@ -26,142 +28,385 @@ func TestGetPreferredUsername(t *testing.T) {
 	}
 }
 
-func TestProvisionNewIdentity(t *testing.T) {
-	expectedProviderName := "myprovidername"
-	expectedProviderUserName := "myusername"
-	expectedUserName := "myusername"
-	expectedUserUID := "myuseruid"
-	expectedIdentityName := "myprovidername:myusername"
-
-	// Expect identity to fully specify a user name and uid
-	expectedCreateIdentity := &api.Identity{
-		ObjectMeta:       kapi.ObjectMeta{Name: expectedIdentityName},
-		ProviderName:     expectedProviderName,
-		ProviderUserName: expectedProviderUserName,
-		User: kapi.ObjectReference{
-			Name: expectedUserName,
-			UID:  types.UID(expectedUserUID),
-		},
-		Extra: map[string]string{},
+func TestProvision(t *testing.T) {
```

Comment:
Split this by the `UserStrategyType`.  If they're really all the same, then having a test case type is fine, but right now this is kind of ridiculous to try to read.